### DISCUSSION TITLE
Problem with port 443 for https

### DISCUSSION THREAD

**Discussion by afr-ux** [rejetto/hfs#440] [Q&A]
I am trying setup https on port 443, I get the following error. I've googled it but I don't know how to fix it:

lacking permission on port 443, try with permission (administrator) or port > 1024

I have Windows v10.0.19045.3803].
I don't have other software running on port 443.
I tried run as administrator, and it works same.

I try cmd (with administrator mode):
net stop winnat
net start winnat
But with first line, it said that this service is not running.

I tried reboot computer too.

Any idea? Thanks.

**Comment by rejetto** [comment]
open the browser on https://localhost e see if there's anything already on it.
I expect nothing, because otherwise hfs should get a different error, but it's worth a try.

**Comment by afr-ux** [comment]
In this computer, port 80 works fine, and port 443 doesn't work.

http://localhost
show user website OK.
https://localhost
say: network error

If I use https in port 81 works fine. But I can see ":81" in url.

In other computer, in same LAN, both ports (80 and 443) work fine.

Thanks.

2024-01-10 15:44 GMT+01:00, Massimo Melina ***@***.***>:

**Comment by rejetto** [comment]
is "network error" the exact message reported by your browser?

**Comment by afr-ux** [comment]
It is in spanish ;-)

This is:

No se puede acceder a este sitio webSe ha restablecido la conexión.
Prueba a:

Comprobar la conexión
Comprobar el proxy y el cortafuegos
Ejecutar Diagnósticos de red de Windows
ERR_CONNECTION_RESET

I checked firewall too.


2024-01-10 16:59 GMT+01:00, Massimo Melina ***@***.***>:

**Comment by rejetto** [comment]
ok, i was looking specifically for that ERR_CONNECTION_REFUSED :)
now i'm quite sure that hfs is correct at NOT saying that the port is used by another program.

**Comment by rejetto** [comment]
you said "checked firewall". Did you try disabling it?

**Comment by PH5HP** [comment]
Hi,

In my case port 443 is working, but I can not access it locally. Only port 80 can be accessed locally. So port 443 works perfectly for outside connections and also the tests in HFS tell me that the port can be reached from the oudside.

I tested if the port 443 was reachable from the outside  myself by using a VPN. That worked for me locally. :-)

Kind regards,
Henk - PH5HP - fileserver.ph5hp.nl

**Comment by afr-ux** [comment]
I have tried disabling the firewall. The result is the same.



2024-01-10 18:01 GMT+01:00, Massimo Melina ***@***.***>:

**Comment by rejetto** [comment]
do you have other "security" software that block hfs ? in case, try disabling it temporarily

**Comment by afr-ux** [comment]
I can't use it either locally or from outside.

2024-01-10 19:01 GMT+01:00, Henk - PH5HP ***@***.***>:

**Reply by PH5HP** [reply]
Hi,

I think in my case it is due to that I host my HFS here on my  home computer and the url fileserver.ph5hp.nl is a subdomain of ph5hp.nl which is on a server of my webside host.

Kind regards,
Henk - PH5HP - fileserver.ph5hp.nl

**Comment by afr-ux** [comment]
No, I don't have other security software...

2024-01-10 19:13 GMT+01:00, Massimo Melina ***@***.***>:

**Comment by rejetto** [comment]
you can try to run the following line in powershell, and see if it says success or failed
> try {$listener = [System.Net.Sockets.TcpListener]443; $listener.Start(); Write-Host "Success"; $listener.Stop(); } catch { Write-Host "Failed: $_"; }

**Comment by afr-ux** [comment]
I do the same. I have a subdomain and via CName registration it is
redirected to a subdomain on duckdns.org. On my computer I have a
small program that updates my IP in duckdns when it changes. This
works perfectly. Tested on port 80. The problem is when I run hfs.exe
as it can't access port 443. Windows doesn't seem to let it and I
don't know why.

HFS ~ HTTP File Server - Copyright 2021-2023, Massimo Melina ***@***.***>
License https://www.gnu.org/licenses/gpl-3.0.txt
started 2024-1-10 19:02:12
version 0.50.6
build 2024-01-02T10:56:57.046Z
cwd C:\Progs\HFS
node v18.5.0
platform win32
pid 19240
config config.yaml
HINT: type "help" for help
http serving on any network : 80
serving on http://192.168.1.50
certificate loaded for sub.domain.es
https lacking permission on port 443, try with permission
(administrator) or port > 1024
>> try specifying a different port, enter this command: config https_port 1080
loading plugin download-counter
loading plugin antibrute
loading plugin vhosting

Thanks & best

2024-01-10 19:17 GMT+01:00, Henk - PH5HP ***@***.***>:

**Comment by afr-ux** [comment]
I run it in PowerShell with administrator access:
Failed: Excepción al llamar a "Start" con los argumentos "0": "Intento
de acceso a un socket no permitido por sus permisos de acceso"

2024-01-10 19:40 GMT+01:00, Massimo Melina ***@***.***>:

**Comment by rejetto** [comment]
While i'm a bit puzzled by your error message (not because of the spanish), on my system it works. So, at least know we know for sure it has nothing to do with hfs, but it's something on your system.

**Comment by afr-ux** [comment]
I try "netstat -ano" in command line and I get this:

Conexiones activas

  Proto  Dirección local          Dirección remota        Estado           PID
  TCP    0.0.0.0:443            0.0.0.0:0              LISTENING       4
  TCP    [::]:443               [::]:0                 LISTENING       4

PID "4" in task managment is "system". But I don't know how can I unblock it.

2024-01-10 21:17 GMT+01:00, Massimo Melina ***@***.***>:

**Comment by rejetto** [comment]
i suggest this
https://stackoverflow.com/a/55377740/646132

```
Open Services
Right click on World Wide Web Publishing Service
Click STOP
To prevent this issue in future :

Go to Properties
Change Startup type to Manual
```

**Comment by afr-ux** [comment]
In the end I managed to solve it. Here is what worked for me.
Using the following line in the command line:
net stop http
Several Windows services are stopped and it is now possible to use port 443.
I have been testing them one by one until I found that the culprit is:
The Routing and Remote Access service
And, by running "services.msc" I was able to stop this service.
Thank you so much!

2024-01-11 16:20 GMT+01:00, Massimo Melina ***@***.***>:
